src/fastpm.c

```diff
@@ -391,29 +391,46 @@ int run_fastpm(FastPMConfig * config, Parameters * prr, MPI_Comm comm) {
 }
 
 static void
-prepare_deltak(FastPMSolver * fastpm, PM * pm, FastPMFloat * delta_k, Parameters * prr, double aout)
+prepare_deltak(FastPMSolver * fastpm, PM * pm, FastPMFloat * delta_k, Parameters * prr, double aout, enum FastPMSpecies species)
```

Comment:
We'll change this function to pass-in the filename of the power spectrum (or NULL), and the filename of the lineark (or NULL). Then the function no longer care about the species. We'll then use the function to generate baryons too.